captum/attr/_core/neuron/neuron_ablation.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+import torch
+
+from ..._utils.attribution import NeuronAttribution, PerturbationAttribution
+from ..._utils.common import _verify_select_column
+from ..._utils.gradient import _forward_layer_eval
+
+from ..feature_ablation import FeatureAblation
+
+
+class NeuronAblation(NeuronAttribution, PerturbationAttribution):
+    def __init__(self, forward_func, layer, device_ids=None):
+        r"""
+        Args:
+
+            forward_func (callable):  The forward function of the model or any
+                          modification of it
+            layer (torch.nn.Module): Layer for which attributions are computed.
+                          Attributions for a particular neuron in the input or output
+                          of this layer are computed using the argument neuron_index
+                          in the attribute method.
+                          Currently, it is assumed that the inputs or the outputs
+                          of the layer, depending on which one is used for
+                          attribution, can only be a single tensor.
+            device_ids (list(int)): Device ID list, necessary only if forward_func
+                          applies a DataParallel model. This allows reconstruction of
+                          intermediate outputs from batched results across devices.
+                          If forward_func is given as the DataParallel model itself,
+                          then it is not necessary to provide this argument.
+        """
+        NeuronAttribution.__init__(self, forward_func, layer, device_ids)
+        PerturbationAttribution.__init__(self, forward_func)
+
+    def attribute(
+        self,
+        inputs,
+        neuron_index,
+        baselines=None,
+        additional_forward_args=None,
+        feature_mask=None,
+        attribute_to_neuron_input=False,
+        ablations_per_eval=1,
+    ):
+        r"""
+            A perturbation based approach to computing neuron attribution,
+            involving replacing each input feature with a given baseline /
+            reference, and computing the difference in the neuron's input / output.
+            By default, each scalar value within
+            each input tensor is taken as a feature and replaced independently. Passing
+            a feature mask, allows grouping features to be ablated together. This can
+            be used in cases such as images, where an entire segment or region
+            can be ablated, measuring the importance of the segment (feature group).
+            Each input scalar in the group will be given the same attribution value
+            equal to the change in target as a result of ablating the entire feature
+            group.
+
```

Comment:
I think I've seen it used as a baseline / comparison in some papers, but not as a primary method. It would be nice to include one though, so if you know of any papers on this, let me know!